File: ios-bridge/ios-bridge/PlaneLine.m
//
//  PlaneLine.m
//  ios-bridge
//
//  Created by Fredrik on 01.04.18.
//  Copyright © 2018 fwcd. All rights reserved.
//

#import <Foundation/Foundation.h>

#import "PlaneLine.h"

@implementation PlaneLine

- (id) initFromX:(float)startX y:(float)startY toX:(float)endX y:(float)endY {
	[self setStartX:startX];
	[self setStartY:startY];
	[self setEndX:endX];
	[self setEndY:endY];
	return self;
}

- (CGPoint) getStart {
	return CGPointMake([self startX], [self startY]);
}

- (CGPoint) getEnd {
	return CGPointMake([self endX], [self endY]);
}

@end

File: ios-bridge/ios-bridge/PlaneLine.swift
//
//  PlaneLine.swift
//  ios-bridge
//
//  Created by Fredrik on 02.04.18.
//  Copyright © 2018 fwcd. All rights reserved.
//

import Foundation
import UIKit

@objc
public class PlaneLine : NSObject {
	public let start: CGPoint
	public let end: CGPoint
	public let color: UIColor
	
	@objc
	public init(fromX startX: Float, y startY: Float, toX endX: Float, y endY: Float, withColor color: UIColor) {
		self.start = CGPoint(x: Double(startX), y: Double(startY))
		self.end = CGPoint(x: Double(endX), y: Double(endY))
		self.color = color
	}
}
